cms/djangoapps/contentstore/tests/test_transcripts_utils.py

```diff
@@ -643,3 +644,133 @@ def test_unicode(self):
+            },
+            lambda **kwargs: {
+                ('12as34', 'en'): 'YOUTUBE_SUBS',
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'YOUTUBE_SUBS',
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm',
+                ],
+            },
+            lambda **kwargs: {
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'HTML5_SUBS',
+        )
+    )
+    @ddt.unpack
+    @patch('xmodule.video_module.transcripts_utils.edxval_api.get_video_transcript')
+    def test_video_transcript_data(self, video_attributes, side_effect, expected_transcript, mock_get_video_transcript):
+        """
+        Tests video transcript data.
+        """
+        video = Mock(**video_attributes)
+        mock_get_video_transcript.side_effect = side_effect
+        transcript = transcripts_utils.get_video_transcript_data(video, 'en')
+        self.assertEqual(transcript, expected_transcript)
+
+    @ddt.data(
+        (
+            {
+                'edx_video_id': '000-000-000',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda *args: {
+                ('000-000-000', None): True,
+                ('12as34', None): False,
+                ('foo', None): False
+            }[args],
+            (False, '000-000-000')
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda *args: {
+                ('12as34', None): True,
+                ('foo', None): False
+            }[args],
+            (True, '12as34'),
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda *args: {
+                ('12as34', None): False,
+                ('foo', None): True
+            }[args],
+            (True, 'foo'),
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm',
+                ],
+            },
+            lambda *args: {
+                ('foo', None): True
+            }[args],
+            (True, 'foo'),
+        )
+    )
+    @ddt.unpack
```

Comment:
@Qubad786 Any good reason for using this dark magic(`lambda *args: {('foo', None): True}[args]`) above in  `ddt` scenarios? Can not we just use `mock_is_transcript_available.return_value`? I would highly recommend to change this.